torch/profiler/strobelight_cli_function_profiler.py

```diff
@@ -0,0 +1,311 @@
+import functools
+import logging
+import os
+import re
+import subprocess
+import time
+from threading import Lock
+from typing import Optional
+
+logger = logging.getLogger("strobelight_cli_function_profiler")
+
+console_handler = logging.StreamHandler()
+formatter = logging.Formatter(
+    "%(name)s, line %(lineno)d, %(asctime)s, %(levelname)s: %(message)s"
+)
+console_handler.setFormatter(formatter)
+
+logger.addHandler(console_handler)
+logger.setLevel(logging.INFO)
+logger.propagate = False
+
+
+class StrobelightCLIProfilerError(Exception):
+    "Raised when an error happens during strobelight profiling"
+
+
+# A function decorator that wraps profile, if no profiler is provided one with
+# default args is created. A function can be annotated as:
+# @strobelight()
+# @strobelight(profiler = StrobelightFunctionProfiler(stop_at_error=True,..))
+# @strobelight(stop_at_error=True,...)
+def strobelight(profiler=None, **kwargs):
+    if not profiler:
+        profiler = StrobelightCLIFunctionProfiler(**kwargs)
+
+    def strobelight_inner(work_function):
+        @functools.wraps(work_function)
+        def wrapper_function(*args, **kwargs):
+            return profiler.profile(work_function, *args, **kwargs)
+
+        return wrapper_function
+
+    return strobelight_inner
+
+
+def pid_namespace_link(pid: Optional[int] = None) -> str:
+    PID_NAMESPACE_PATH = "/proc/{}/ns/pid"
+    """Returns the link to the process's namespace, example: pid:[4026531836]"""
+    pid = pid or os.getpid()
+    return os.readlink(PID_NAMESPACE_PATH.format(pid))
+
+
+def pid_namespace(pid: Optional[int] = None) -> int:
+    """Returns the process's namespace id"""
+    pid = pid or os.getpid()
+    link = pid_namespace_link(pid)
+    return int(link[link.find("[") + 1 : -1])
+
+
+def command_to_string(command):
+    s = ""
+    for item in command:
+        s += " "
+        s += item
+    return s
```

Comment:
nit: this whole thing is just this and can be inlined:
```
return " " + " ".join(command)
```
(or if you didn't mean to always have a leading space then:
```
return " ".join(command)
```
)